File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO1/DecoratorTest/src/CarFactory.java
import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;

public class CarFactory
{
    private static interface CarMaker { public Car make(); }
    private static final HashMap<String,CarMaker> factory= new HashMap<String,CarMaker>()
    {
        {
            put("Volvo",() -> new Volvo());
            put("Audi",() -> new Audi());
            put("Toyota",() -> new Toyota());
            put("Seat",() -> new Seat());
        }
    };
    public static final List<String> models=factory.keySet().stream().sorted().collect(Collectors.toList());
    public static Car make(String model)
    {
        return factory.get(model).make();
    }
}

File: semestr-8/Jezyk C#/alexander/JCS/2023s/IZ08IO2/DecoratorProject/DecoratorProject/CarFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DecoratorProject
{
    abstract class CarFactory 
    {
        private delegate Car maker();
        private static readonly Dictionary<string, maker> map = new Dictionary<string, maker>()
        {
            {"Volvo",() => new Volvo() },
            {"Seat",() => new Seat() },
            {"Toyota",() => new Toyota() },
            {"Audi",() => new Audi() },
        };
        public static readonly string[] Models = map.Keys.OrderBy(m => m).ToArray();
        public static Car make(string model) { return map[model](); }
    }
}
